### ISSUE TITLE
Extend TestEngine so that it can be used for "single-dataset" engines

### ISSUE THREAD

**Issue by nathanfranklin** [simphony/simphony-common#221]
In `TestEngine`,  `test_iter_datasets` and `test_remove_dataset` add 5 datasets for testing purposes.  Some engines only support a **single** dataset.

**Comment by nathanfranklin** [comment]
`TestEngine` could be given given a method that could be overridden for testing *single-dataset* engines

something like:
```
    def number_datasets_used_in_testing(self):
        """ Number of datasets used in testing engine

        """
        return 5
```

**Comment by nathanfranklin** [comment]
Closed by #224
